File: client/src/app/student-management/interface/Student.ts
export interface Student {
    studentId: string;
    fullName: string;
    dob: string;
    gender: string;
    faculty: string;
    intake: string;
    program: string;
    status: string;
    permanentAddress: string;
    temporaryAddress: string;
    email: string;
    phone: string;
    documentType: string;
    documentNumber: string;
    issuedDate: string;
    expiredDate: string;
    issuedBy: string;
    issuedCountry: string;
    hasChip: boolean;
    nationality: string;
}

export const defaultStudent: Student = {
    studentId: '',
    fullName: '',
    dob: '',
    gender: '',
    faculty: '',
    intake: '',
    program: '',
    status: '',
    permanentAddress: '',
    temporaryAddress: '',
    email: '',
    phone: '',
    documentType: '',
    documentNumber: '',
    issuedDate: '',
    expiredDate: '',
    issuedBy: '',
    issuedCountry: '',
    hasChip: false,
    nationality: ''
};

File: server/src/main/java/org/example/backend/domain/Student.java
package org.example.backend.domain;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.example.backend.common.Auditable;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "students")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Student extends Auditable {
    @Id
    @Column(name = "student_id", length = 10)
    private String studentId;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(name = "dob", nullable = false)
    private LocalDate dob;

    @Column(name = "gender", nullable = false)
    private String gender;

    @Column(name = "intake")
    private String intake;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "phone_country", nullable = false)
    private String phoneCountry;

    @Column(name = "phone", nullable = false, unique = true)
    private String phone;

    @Column(name = "nationality")
    private String nationality;

    @ManyToOne
    @JoinColumn(name = "faculty_id")
    private Faculty faculty;

    @ManyToOne
    @JoinColumn(name = "program_id")
    private Program program;

    @ManyToOne
    @JoinColumn(name = "student_status_id")
    private StudentStatus studentStatus;

    @OneToMany(mappedBy = "student", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Address> addresses;

    @OneToMany(mappedBy= "student", cascade= CascadeType.ALL, orphanRemoval = true)
    private List<Document> documents;

    @OneToMany(mappedBy = "student")
    private List<ClassRegistration> classRegistrations;
}
